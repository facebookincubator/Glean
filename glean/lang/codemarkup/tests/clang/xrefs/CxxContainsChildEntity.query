query: codemarkup.cxx.CxxContainsChildEntity _
transform: [gensort, []]
