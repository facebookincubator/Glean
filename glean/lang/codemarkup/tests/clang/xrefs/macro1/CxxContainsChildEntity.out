[
  "@generated",
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 19,
                "columnBegin": 1,
                "lineEnd": 23,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "bar" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N" } } }
                  }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "void" },
                  "parameters": [
                    { "name": { "key": "" }, "type": { "key": "int" } }
                  ]
                }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 21,
                "columnBegin": 3,
                "lineEnd": 21,
                "columnEnd": 18
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 19,
                "columnBegin": 1,
                "lineEnd": 23,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N" } } }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 20,
                "columnBegin": 3,
                "lineEnd": 20,
                "columnEnd": 15
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "namespace_": {
            "key": {
              "name": { "key": { "name": { "key": "N" } } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 19,
                "columnBegin": 1,
                "lineEnd": 23,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "var" },
                  "scope": {
                    "namespace_": { "key": { "name": { "key": "N" } } }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": {
                "global_": { "kind": 0, "attribute": 0, "definition": true }
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 22,
                "columnBegin": 3,
                "lineEnd": 22,
                "columnEnd": 9
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Foo" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 25,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "bar" } } },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": {
                  "returns": { "key": "void" },
                  "parameters": [
                    { "name": { "key": "" }, "type": { "key": "int" } }
                  ]
                }
              },
              "method": {
                "isVirtual": false,
                "isConst": false,
                "isVolatile": false,
                "refQualifier": 0
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 27,
                "columnBegin": 3,
                "lineEnd": 27,
                "columnEnd": 18
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Foo" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 25,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "function_": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": { "name": { "key": "foo" } } },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "signature": {
                "key": { "returns": { "key": "void" }, "parameters": [ ] }
              },
              "method": {
                "isVirtual": false,
                "isConst": false,
                "isVolatile": false,
                "refQualifier": 0
              },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 26,
                "columnBegin": 3,
                "lineEnd": 26,
                "columnEnd": 15
              }
            }
          }
        }
      }
    }
  },
  {
    "key": {
      "parent": {
        "decl": {
          "record_": {
            "key": {
              "name": {
                "key": { "name": { "key": "Foo" }, "scope": { "global_": { } } }
              },
              "kind": { "struct_": { } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 25,
                "columnBegin": 1,
                "lineEnd": 29,
                "columnEnd": 1
              }
            }
          }
        }
      },
      "child": {
        "decl": {
          "variable": {
            "key": {
              "name": {
                "key": {
                  "name": { "key": "field" },
                  "scope": {
                    "recordWithAccess": {
                      "record": {
                        "key": {
                          "name": { "key": "Foo" },
                          "scope": { "global_": { } }
                        }
                      },
                      "access": 0
                    }
                  }
                }
              },
              "type": { "key": "int" },
              "kind": { "field": { "mutable_": false } },
              "source": {
                "file": { "key": "test.cpp" },
                "lineBegin": 28,
                "columnBegin": 3,
                "lineEnd": 28,
                "columnEnd": 11
              }
            }
          }
        }
      }
    }
  }
]